spec/pg/connection_spec.rb

```diff
@@ -893,15 +890,36 @@
 		expect( conn.connect_poll ).to eq( PG::PGRES_POLLING_FAILED )
 	end
 
-	it "discards previous results (if any) before waiting on an #async_exec" do
+	it "discards previous results at #discard_results" do
 		@conn.send_query( "select 1" )
-		res = @conn.async_exec( "select 42 as one" )
+		@conn.discard_results
+		@conn.send_query( "select 41 as one" )
+		res = @conn.get_last_result
+		expect( res.to_a ).to eq( [{ 'one' => '41' }] )
+	end
+
+	it "discards previous results (if any) before waiting on #exec and #async_exec" do
+		@conn.send_query( "select 1" )
+		res = @conn.exec( "select 42 as one" )
 		expect( res.to_a ).to eq( [{ 'one' => '42' }] )
 	end
 
-	it "calls the block if one is provided to #async_exec" do
+	it "discards previous errors before waiting on #exec and #async_exec", :without_transaction do
+		@conn.send_query( "ERROR" )
+		res = @conn.exec( "select 43 as one" )
+		expect( res.to_a ).to eq( [{ 'one' => '43' }] )
+	end
+
+	it "discards previous cancels before waiting on #exec and #async_exec", :without_transaction do
+		@conn.send_query( "SEELCT pg_sleep(5)" )
```

Comment:
Is this typo intentional?

Reply:
:smile: Removed the test case - it was a duplication anyway.